autoload/neoterm/repl.vim

```diff
@@ -70,6 +80,29 @@ function! neoterm#repl#opfunc(type)
   call g:neoterm.repl.exec(l:lines)
 endfunction
 
+function! neoterm#repl#cell()
+  let l:lnum1 = search(g:neoterm_repl_cellmarker, 'Wb') + 1
+  let l:lnum2 = search(g:neoterm_repl_cellmarker, 'W') - 1
+  if l:lnum2 == -1
+    let l:lnum2 = line('$')
+  endif
+  let l:lines = getline(l:lnum1, l:lnum2)
+  call g:neoterm.repl.exec(l:lines)
+endfunction
+
+function! neoterm#repl#run()
+  try
+    call g:neoterm.repl.instance().exec(add([g:neoterm_repl_run], g:neoterm_eof))
+  catch
+     echo "The filetype is currently not supported!"
+  endtry
+endfunction
+
 function! g:neoterm.repl.exec(command)
-  call g:neoterm.repl.instance().exec(add(a:command, g:neoterm_eof))
```

Comment:
It seems to me that python, requires a unique way to `exec` the commands. To make that viable we can create a dynamic call to `neoterm#repl#python#exec(command)` function, based on the filetype. This way we wouldn't need more options, which leads to less code, and also we would have a good structure if any other language have special needs.

For that, we could change the `g:neoterm.repl.exec` to try to exec a `neoterm#repl#<filetype>#exec`, if the function doesn't exist it keepts the current behavior:

```vimscript
function! g:neoterm.repl.exec(command)
  let l:ft_exec = printf('neoterm#repl#%s#exec', &filetype)
  let l:cmd = add(a:command, g:neoterm_eof)

  try
    let ExecByFiletype = function(l:ft_exec)
    call ExecByFiletype(l:cmd)
  catch /^Vim\%((\a\+)\)\=:E117/
    call g:neoterm.repl.instance().exec(l:cmd)
  endtry
endfunction
```

With that in place, you could than create a file in  `autoload/neoterm/repl/python.vim` with a function like:
```vimscript
function! neoterm#repl#python#exec(command)
  call setreg('+', a:command, 'l')
  call g:neoterm.repl.instance().exec(add(["%paste"], g:neoterm_eof))
endfunction
```

Reply:
Thast a good idea :)